yiisoft/yii2-elasticsearch
# Elasticsearch 5.x support

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes/no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #112

Notes:
- removed Query::filter()
- queries can be set through either `Query::query()` or through `Query::where()`
- rewrite `QueryBuilder` condition building to support Elasticsearch 5.0. All the where params are wrapped into `constant_score` query
- optimized `Query::andWhere()` and `Query::orWhere()` so it won't generate unnecessary subqueries on subsequent calls.
- remove `Query::fields()` (fields query part is removed in favor of stored_fields and works differently in 5.0)
- add `Query::storedFields()`, `Query::scriptFields()`
- reset`CURL_POSTFIELDS` in `Connection::resetCurlHandle()`, it was causing test errors in 5.4 (because in some cases requests with body and subsequent requests without it were not cleared)
- remove `Command::deleteMapping` because mapping deletion is no longer supported in Elasticsearch 5.0. You can only delete and recreate full index.

It breaks BC with previous version of extension. So I don't know, maybe there has to be two different versions of extension for different elasticsearch versions or it has to be refactor in some way to support running both Elasticsearch >=5.0 and <5.0 by switch
